docs-ui/components/code.tsx

```diff
@@ -0,0 +1,145 @@
+import 'prismjs/themes/prism.css';
+
+import {createRef, RefObject, useEffect, useState} from 'react';
+import {withTheme} from '@emotion/react';
+import styled from '@emotion/styled';
+import copy from 'copy-text-to-clipboard';
+import Prism from 'prismjs';
+
+import {IconCode} from 'app/icons';
+import space from 'app/styles/space';
+import {Theme} from 'app/utils/theme';
+
+type Props = {
+  theme: Theme;
+  /**
+   * Main code content gets passed as the children prop
+   */
+  children: string;
+  /**
+   * Auto-generated class name for <pre> and <code> element,
+   * with a 'language-' prefix, e.g. language-css
+   */
+  className?: string;
+  /**
+   *  Meta props from the markdown syntax,
+   *  for example, in
+   *
+   * ```jsx label=hello
+   * [some code]
+   * ```
+   *
+   * the label prop is set to 'hello'
+   */
+  label?: string;
+};
+
+const Code = ({theme, children, className, label}: Props) => {
+  const codeRef: RefObject<HTMLElement> = createRef();
+  const copyButtonRef: RefObject<HTMLButtonElement> = createRef();
+
+  const [copied, setCopied] = useState(false);
+
+  function handleCopyCode() {
+    const copyButton = copyButtonRef?.current;
+
+    if (!copyButton) {
+      return;
+    }
+
+    /** Remove comments from code */
```

Comment:
Avoid using flag comments unless they are multi-line, and are documenting something like a function or module level variable

Can just be


```tsx
// Remove comments from code
```